### DISCUSSION TITLE
As a 'desktop app' is the 'Shutdown' button/function a feature or flaw?

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1082] [General]
amiberry v5.6.1 / x86-64 

Just holding this up as in comparison, to using amiberry as a desktop application, in the same way one would use ie; the fs-uae-launcher application (or just about any other desktop app), in that you have a 'Quit' button (close prog), and not a system wide Shutdown button ~ shutdown is handled by display/window manager or other input events. 

That said, when amiberry is used in a dedicated Amiga emulation setup (like I'm building on x86-64 thin-client machines), the 'Shutdown' button/function can absolutely be considered an (almost vital) feature, and in this usage context the 'Shutdown' button/function should imho always be there.

If one looks at this, from ie; a debian packager's viewpoint, the Shutdown button/function relies upon current user being in the sudoers list, and if not so, it will fail ... and semantically perhaps, one shouldn't need consider this wrt a binary package build... I mean, it is what it is yeah?...user might invoke SynapticPM to install amiberry (or cmdline equiv), end of the day they don't expect (any?) an Amiga emulation desktop application to provided them with a system shutdown function...ie; "It's just an app, not a power manager" =)

I'm just wondering what's the correct 'etiquette' here? For instance, does this fall within the packager's purview, to provide their own patch (remove shutdown widget from GUI code, and move quit widget to it's location), or would this be considered an enhancement to amiberry itself ...ie; achieve the same ends using a compile time env_var & diversion in the code?

**Comment by midwan** [comment]
There's already an option to disable the Shutdown button, if you don't want it in some environments. Check `amiberry.conf` ;-)

**Comment by giantclambake** [comment]
My bad ~ I didn't think to check the conf file =)

Thanks for the answer.

**Comment by giantclambake** [comment]
M'kay.... my 6+ decade old eyes may have missed the point, but now that I know that option is there in amiberry.conf, having toggled the option and seen the resultant GUI, we end up at the (subjective, visual) semantics wrt to the words 'disable' and 'disallow'. Keep in mind I'm merely speaking to amiberry's 'presentation' as a desktop app...

... if you set  ' disable_shutdown_button=yes' in amiberry.conf you get a GUI that looks like this...

![Screenshot of current behavior](https://i.imgur.com/xasj4Ok.png)

....on the other hand, if you had a compile time env_var of ie; enable-shutdown={yes/no} [defaults to yes if unset] , you can...

![Screenshot of my preferred behavior](https://i.imgur.com/5SVOp0k.png)

As said, such things are bound to be subjective, but in presentation what I see is the difference between amiberry being 'designed/catering for' desktop app usage, as opposed to (obviously) being something else, with something (visibly) disabled for desktop app usage. For mine, not having the Shutdown button there, allows the Quit button to be moved bottom-left, providing those GUI niceties like a good physical/visible distance between  those buttons used to interact with the emulation, and the Quit button itself (so it's far enough away from 'sloppy mousing' =)

I suppose that brings me full circle in a way ...ie; if I wanted to implement changes like that above, what's the best way forward to achieve that ~ request an amiberry enhancement for this, or just hack&patch my way into oblivion  8^)
